compiler/parser/RdrHsSyn.hs

```diff
@@ -2844,17 +2873,18 @@ mkTemplateDecls (L _ th@(TemplateHeader _ lname@(L nloc _) tyVars)) fields (L _
                            (mkUnqualVar $ mkVarOcc "this")
       , ccdChoiceData = ChoiceData {
             cdChoiceName = noLoc $ qualifyDesugar $ mkTcOcc "Archive"
-          , cdChoiceFields = Nothing
+          , cdChoiceFields = noLoc $ HsRecTy noExt []
           , cdChoiceReturnTy = unitType
           , cdChoiceBody = pureUnit
           , cdChoiceConsuming = noLoc $ Just PreConsuming
           , cdChoiceDoc = Nothing
           }
+      , ccdTypeVars = []
```

Comment:
What is stopping us from computing this list here?

Reply:
This is the list of type vars for the `Archive` choice, which should be empty. But you're right that I should probably compute the type var list when initially constructing the choice data in `validateTemplateBodyDecls`.